### PR TITLE
sqlod changes for cloud servertype

### PR DESCRIPTION
**Pull request by alljamziMS**
The ServerType of SQL OD in SMO is being changed to cloud. This is a minimal set of changes needed in sqltoolservice in order to keep everything working within ADS

### CODE REVIEW

File: src/Microsoft.SqlTools.ServiceLayer/Connection/ListDatabaseRequestHandler.cs [hunk 1]
```diff
@@ -25,9 +26,9 @@ public class ListDatabasesRequestDatabaseProperties
     /// </summary>
     static class ListDatabaseRequestHandlerFactory
     {
-        public static IListDatabaseRequestHandler getHandler(bool includeDetails, bool isSqlDB)
+        public static IListDatabaseRequestHandler getHandler(bool includeDetails, bool isSqlDB, ConnectionInfo connectionInfo = null)
         {
-            if (!includeDetails)
+            if (!includeDetails || (connectionInfo != null && connectionInfo.EngineEdition == DatabaseEngineEdition.SqlOnDemand))
```
Comment (Charles-Gagnon) [inline src/Microsoft.SqlTools.ServiceLayer/Connection/ListDatabaseRequestHandler.cs:31]: `connectionInfo?.EngineEdition == DatabaseEngineEdition.SqlOnDemand`

**Review by Charles-Gagnon — CHANGES_REQUESTED** [review]
Don't you need to update the SMO version too?

File: src/Microsoft.SqlTools.ServiceLayer/Connection/ListDatabaseRequestHandler.cs (hunk 1 above)
Comment (BranislavGrbicMDCS) [reply src/Microsoft.SqlTools.ServiceLayer/Connection/ListDatabaseRequestHandler.cs:31]: @Charles-Gagnon - there is another STS change on the way. 
I'll be merging the changes so there is only one review.

The comment is resolved in the PR that is on the way.

**Comment by Charles-Gagnon** [comment]
@BranislavGrbicMDCS @alljamziMS Are you planning on finishing this at some point? If not it'd be great if you would close this to clean up stale PRs

**Pull request closed without merging**